src/EmbedIO/Sessions/ISessionManager.cs

```diff
@@ -33,6 +33,13 @@ public interface ISessionManager
         /// <param name="id">The unique ID of the session.</param>
         /// <seealso cref="ISession.Id"/>
         void Delete(IHttpContext context, string id);
+        
+        /// <summary>
+        /// Deletes the session (if any) associated with the specified context.
+        /// </summary>
+        /// <param name="context">The HTTP context.</param>
+        /// <seealso cref="ISession.Id"/>
+        void Delete(IHttpContext context);
```

Comment:
This is a [binary breaking change](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/breaking-changes). According to semantic versioning best practices, it requires a major version increase. This in turn comes at a non-negligible cost: package compatibility, documentation, and so on.

I'm not saying that it should not be done (nor that it should; I'm still in mid-review), but if this can be postponed a bit, and other useful breaking changes thrown in the mix, the major version increase would make a lot more sense.

Besides, this method could probably replace the other `Delete` method entirely, as long as existing EmbedIO code is concerned. The impact on user code would most probably be close to zero, as no alternative session managers exist AFAIK. To tell the truth, I don't see any possible use case for `Delete(context, id)`: the public interface of a session manager always refers to an HTTP context, so there should be no need of addressing any other session than the one associated with the context.

_"So why did you put `Delete(context, id)` here in the first place?"_ I frankly don't remember. Maybe because there were both `DeleteSession(context)` and `DeleteSession(session)` in v2 and I thought there may have been an actual need of deleting a specific, not context-related session. Looking at it now, keeping `Delete(context)` and removing `Delete(context, id)` makes more sense to me.